src/hugr/views/sibling.rs

```diff
@@ -0,0 +1,562 @@
+//! Views for HUGR sibling subgraphs.
+//!
+//! Views into convex subgraphs of HUGRs within a single level of the
+//! hierarchy, i.e. within a sibling graph. Convex subgraph are always
+//! induced subgraphs, i.e. they are defined by a subset of the sibling nodes.
+//!
+//! Sibling subgraphs complement [`super::HierarchyView`]s in the sense that the
+//! latter provide views for subgraphs defined by hierarchical relationships,
+//! while the former provide views for subgraphs within a single level of the
+//! hierarchy.
+
+use itertools::Itertools;
+use portgraph::{algorithms::ConvexChecker, view::Subgraph, Direction, PortView};
+use thiserror::Error;
+
+use crate::{
+    ops::{
+        handle::{ContainerHandle, DataflowOpID},
+        OpTag, OpTrait,
+    },
+    types::FunctionType,
+    Hugr, Node, Port, SimpleReplacement,
+};
+
+use super::{sealed::HugrInternals, HugrView};
+
+/// A non-empty convex subgraph of a HUGR sibling graph.
+///
+/// A HUGR region in which all nodes share the same parent. Unlike
+/// [`super::SiblingGraph`],  not all nodes of the sibling graph must be
+/// included. A convex subgraph is always an induced subgraph, i.e. it is defined
+/// by a set of nodes and all edges between them.
+
+/// The incoming boundary (resp. outgoing boundary) is given by the input (resp.
+/// output) ports of the subgraph that are linked to nodes outside of the subgraph.
+/// The signature of the subgraph is then given by the types of the incoming
+/// and outgoing boundary edges. Given a replacement with the same signature,
```

Comment:
nit: `edges` -> `ports`? If I understand correctly and you include a node with a port that has many edges to outside the subgraph, the port gets included once, not once per edge?